src/cli.py

```diff
@@ -78,6 +92,22 @@ def deserialize_scope(cls, v: str | list[str]) -> str:
     def serialize_scope(self, scope: str) -> list[str]:
         return scope.split()
 
+    @field_validator("metadata", mode="before")
+    @classmethod
+    def deserialize_metadata(cls, v: str | dict[str, Any]) -> dict[str, Any]:
+        if isinstance(v, dict):
+            return v
+
+        kv = v.split(",")
+
+        metadata = {}
+
+        for pair in kv:
+            key, value = pair.split("=")
+            metadata[key] = value
```

Comment:
nit(feel free to ignore): slightly more pythonic
```python
metadata = {k: v for kv in value.split(",") for k, v in kv.split("=")}
```

Reply:
happy to create another issue for this